### PR TITLE
SLI-901 Let core register smart notifications

### PR DESCRIPTION
**Pull request by nquinquenel**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt [hunk 1]
```diff
@@ -177,6 +174,15 @@ object SonarLintIntelliJClient : SonarLintClient {
         showBalloon(null, params.text, convert(params.type))
     }
 
+    override fun showSmartNotification(params: ShowSmartNotificationParams) {
+        val projects = params.scopeIds.stream()
+            .map { BackendService.findModule(it)?.project ?: BackendService.findProject(it) }
+            .filter { it != null}
```
Comment (damien-urruty-sonarsource) [inline src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt:180]: You can combine map and filter with `mapNotNull`

File: src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt [hunk 2]
```diff
@@ -177,6 +174,15 @@ object SonarLintIntelliJClient : SonarLintClient {
...
+            .collect(Collectors.toSet())
```
Comment (damien-urruty-sonarsource) [inline src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt:181]: I think you can replace that by `toSet()`

File: src/main/java/org/sonarlint/intellij/notifications/SonarLintProjectNotifications.java [hunk 3]
```diff
@@ -133,4 +140,34 @@ public void expireCurrentHotspotNotificationIfNeeded() {
       currentOpenHotspotNotification = null;
     }
   }
+
+  public void handle(ShowSmartNotificationParams smartNotificationParams) {
+    var telemetry = getService(SonarLintTelemetry.class);
+    var category = smartNotificationParams.getCategory();
+    telemetry.devNotificationsReceived(category);
```
Comment (damien-urruty-sonarsource) [inline src/main/java/org/sonarlint/intellij/notifications/SonarLintProjectNotifications.java:147]: Isn't this managed by the backend already ?

**Review by damien-urruty-sonarsource — APPROVED** [review]
A few comments, else LGTM

File: src/main/java/org/sonarlint/intellij/notifications/SonarLintProjectNotifications.java (hunk 3 above)
Comment (nquinquenel) [reply src/main/java/org/sonarlint/intellij/notifications/SonarLintProjectNotifications.java]: Yes you're right, only dev notifications clicked should be counted on client side

File: src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt (hunk 1 above)
Comment (nquinquenel) [reply src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt]: Thanks, I didn't know!

**Review by damien-urruty-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**